concrete/src/Application/UserInterface/Sitemap/StandardSitemapProvider.php

```diff
@@ -143,22 +142,32 @@ public function getRequestedSiteTree()
             $this->cookieJar->getResponseCookies()->addCookie($cookieKey, $query->get('siteTreeID'));
 
             return $this->siteService->getSiteTreeByID($query->get('siteTreeID'));
-        } elseif ($this->cookieJar->has($cookieKey)) {
-            return $this->siteService->getSiteTreeByID($this->cookieJar->get($cookieKey));
         } else {
-            $site = $this->siteService->getActiveSiteForEditing();
-            $locale = $site->getDefaultLocale();
-            if ($locale && $this->checkPermissions($locale)) {
-                return $locale->getSiteTreeObject();
+            // Check if the site id in $cookieKey is valid
+            $site = null;
+            if ($this->cookieJar->has($cookieKey)) {
+                $site = $this->siteService->getSiteTreeByID($this->cookieJar->get($cookieKey));
             }
+            if (is_object($site)) {
```

Comment:
Should we add a comment that the site that was stored in the cookie may have been deleted and that's why we again check if `$site` is a valid object?